src/test/java/co/rsk/bitcoinj/script/FlyoverRedeemScriptParserTest.java

```diff
@@ -0,0 +1,291 @@
+import co.rsk.bitcoinj.core.Coin;
+import co.rsk.bitcoinj.core.NetworkParameters;
+import co.rsk.bitcoinj.core.Sha256Hash;
+import co.rsk.bitcoinj.core.VerificationException;
+import co.rsk.bitcoinj.crypto.TransactionSignature;
+import co.rsk.bitcoinj.params.MainNetParams;
+import co.rsk.bitcoinj.script.RedeemScriptParser.MultiSigType;
+import java.math.BigInteger;
+import java.util.List;
+import org.junit.Before;
+import org.junit.Test;
+
+public class FlyoverRedeemScriptParserTest {
+
+    private final List<BtcECKey> defaultRedeemScriptKeys = RedeemScriptUtils.getDefaultRedeemScriptKeys();
+    private final List<BtcECKey> emergencyRedeemScriptKeys = RedeemScriptUtils.getEmergencyRedeemScriptKeys();
+    private final Sha256Hash derivationArgumentsHash = Sha256Hash.of(new byte[]{1});
+    private Script standardRedeemScript;
+    private Script flyoverStandardRedeemScript;
+    private Script erpRedeemScript;
+    private Script flyoverErpRedeemScript;
+    private Script p2shErpRedeemScript;
+    private Script flyoverP2shErpRedeemScript;
+
+    @Before
+    public void setUp() {
+        final long CSV_VALUE = 52_560L;
+        standardRedeemScript = RedeemScriptUtils.createStandardRedeemScript(defaultRedeemScriptKeys);
+        flyoverStandardRedeemScript = RedeemScriptUtils.createFlyoverRedeemScript(derivationArgumentsHash.getBytes(), standardRedeemScript);
+
+        erpRedeemScript = RedeemScriptUtils.createErpRedeemScript(defaultRedeemScriptKeys, emergencyRedeemScriptKeys, CSV_VALUE);
+        flyoverErpRedeemScript = RedeemScriptUtils.createFlyoverRedeemScript(derivationArgumentsHash.getBytes(), erpRedeemScript);
+
+        p2shErpRedeemScript = RedeemScriptUtils.createP2shErpRedeemScript(defaultRedeemScriptKeys, emergencyRedeemScriptKeys, CSV_VALUE);
+        flyoverP2shErpRedeemScript = RedeemScriptUtils.createFlyoverRedeemScript(derivationArgumentsHash.getBytes(), p2shErpRedeemScript);
+    }
+
+    @Test
+    public void getMultiSigType_whenIsStandardRedeemScript_shouldReturnFlyoverMultiSigType() {
+        // Arrange
+        FlyoverRedeemScriptParser flyoverRedeemScriptParser = new FlyoverRedeemScriptParser(flyoverStandardRedeemScript.getChunks());
+
+        // Act
+        MultiSigType actualMultiSigType = flyoverRedeemScriptParser.getMultiSigType();
+
+        // Assert
+        assertEquals(MultiSigType.FLYOVER, actualMultiSigType);
+    }
+
+    @Test
+    public void getMultiSigType_whenIsErpRedeemScript_shouldReturnFlyoverMultiSigType() {
+        // Arrange
+        FlyoverRedeemScriptParser flyoverRedeemScriptParser = new FlyoverRedeemScriptParser(flyoverErpRedeemScript.getChunks());
+
+        // Act
+        MultiSigType actualMultiSigType = flyoverRedeemScriptParser.getMultiSigType();
+
+        // Assert
+        assertEquals(MultiSigType.FLYOVER, actualMultiSigType);
+    }
+
+    @Test
+    public void getMultiSigType_whenIsP2shErpRedeemScript_shouldReturnFlyoverMultiSigType() {
```

Comment:
Are some test cases passing malformed and other types of script missing? Wdyt?

Reply:
For `getMultiSigType` method, we created tests for`StandardRedeemScript`, `ErpRedeemScript`, and `P2shErpRedeemScript` scripts. They correspond to every single script related to the `FastBridge` classes. Should we test  additional scripts?

Reply:
Well, let me create a new task to address this recommendation after the `RedeemScriptParserFactory` refactor is merged. I'll be working on it, invalid script cases, and edge cases as well.